packages/eas-cli/src/submissions/utils/builds.ts

```diff
@@ -0,0 +1,80 @@
+import gql from 'graphql-tag';
+
+import { graphqlClient } from '../../api';
+import { SubmissionPlatform } from '../types';
+
+const graphqlPlatform: Record<SubmissionPlatform, string> = {
+  [SubmissionPlatform.Android]: 'ANDROID',
+  [SubmissionPlatform.iOS]: 'IOS',
+};
+
+export async function getBuildArtifactUrlByIdAsync(
+  platform: SubmissionPlatform,
+  buildId: string
+): Promise<string> {
+  const { data, error } = await graphqlClient
+    .query(
+      gql`
+    {
+      builds {
+        byId(buildId: "${buildId}") {
+          platform,
+          artifacts {
+            buildUrl
+          }
+        }
+      }
+    }`
+    )
+    .toPromise();
+
+  if (error?.graphQLErrors) {
+    throw error.graphQLErrors[0];
+  }
+
+  const {
+    platform: buildPlatform,
+    artifacts: { buildUrl },
+  } = data.builds.byId;
+
+  if (buildPlatform !== graphqlPlatform[platform]) {
+    throw new Error("Build platform doesn't match!");
+  }
```

Comment:
```suggestion
```
I think we don't want to check that. If something like this happened it would mean that our API is broken.

Reply:
We have to, because we cannot filter it in gql `builds.byId` query. This is called when user specifies build ID on his own - he may copy invalid ID by mistake.

Reply:
Ah, I see